### ISSUE TITLE
Dynamic import does not work in all browsers

### ISSUE THREAD

**Issue by connorjclark** [emscripten-core/emscripten#16625]
TLDR: Dynamic import does not work in all browsers, so usage of ES6 imports in workers to support pthreads is problematic

When targeting an `.mjs` output, emscripten will export JS as ES6. This mostly works, but FF does not yet support ES6 imports in workers, so I get this error in FF:

```
Error: Dynamic module import is disabled or not supported in this context 
```
(from module.worker.js)

If this is a known limitation, perhaps it should be noted [here](https://emscripten.org/docs/tools_reference/emcc.html?highlight=emcc#:~:text=ES6%20JavaScript%20module).

For anyone wanting to support more than just Chrome this makes ".mjs" a non-starter when using pthreads. If pthreads workers can't be implemented without relying on ES6, perhaps emscripten could emit a big warning when this happens warning that these flags are currently incompatible with cross-browser support?

ref https://bugzilla.mozilla.org/show_bug.cgi?id=1247687

**Comment by sbc100** [comment]
When you output to `.mjs` you are effectively opt-int into `-sEXPORT_ES6` and `-sMODULARIZE` .

It sounds like you are saying that in this mode, pthreads are not well supported on all browsers.

If so, I agree we should issue a warning for now, and perhaps look into addressing this issue in some other way. 

BTW, is it just `-sEXPORT_ES6` that is the issue or does `-sMODULARIZE` on its own trigger this issue?

**Comment by kg** [comment]
The code in question requires both export and modularize:

```
#if MODULARIZE && EXPORT_ES6
      (e.data.urlOrBlob ? import(e.data.urlOrBlob) : import('./{{{ TARGET_JS_NAME }}}')).then(function(exports) {
```

We're hitting this trying to combine modules and es6 with workers, and it's not obvious how to work around it since (afaik) this is a SyntaxError that occurs at parse time instead of an error that occurs at runtime, so hiding it inside an if statement won't work.

Since it's running in a worker, could emscripten in the main thread do the module import on behalf of the worker somehow?

**Comment by DavidPeicho** [comment]
There is an open bug on Firefox: https://bugzilla.mozilla.org/show_bug.cgi?id=1540913
The goal would be to support dynamic module in worker's context.

I think it should be possible for emscripten to directly load the JS code and eventually execute a load later on?
